src/App.tsx

```diff
@@ -269,6 +269,15 @@ class App extends Component {
                                     await Codap.getDataContext(selectedDataContext);
 
         if (!existingDataContext) {
+          // map parent collection names
+          const collectionNames: string[] = [];
+          (sharedDataContext.collections || [])
+            .forEach((collection, index) => {
+              collection && collectionNames.push(collection.name);
+              if (collection.parent) {
+                collection.parent = collectionNames[index - 1];
+              }
+            });
```

Comment:
I don't understand the purpose of this. Shouldn't every collection have a correct reference to its parent, by parent name, as set when the data context is processed in `getSharableDataContext`? If there's a place where that's not being called, and data contexts are being shared with parent ids instead, we should fix that.